pulser/simulation/simulation.py

```diff
@@ -700,11 +781,56 @@ def build_coeffs_ops(basis: str, addr: str) -> list[list]:
             self.operators[addr][basis] = operators
             return terms
 
-        # Time independent term:
+        def build_masked_coeff() -> np.ndarray:
+            coeff = np.zeros(self._tot_duration)
+            try:
+                ti = self._seq._slm_mask_time[0]
+                tf = self._seq._slm_mask_time[1]
+            except IndexError:
+                return coeff
+            coeff[ti:tf] = 1
+            return coeff
+
+        def build_unmasked_coeff() -> np.ndarray:
+            coeff = np.ones(self._tot_duration)
+            try:
+                ti = self._seq._slm_mask_time[0]
+                tf = self._seq._slm_mask_time[1]
+            except IndexError:
+                return coeff
+            coeff[ti:tf] = 0
+            return coeff
```

Comment:
Two things:

1. One seems to be just the logical NOT of the other
2. Why do you need these in functions?

Reply:
1. True, I could just leave one
2. I thought it looked more neat and clearer. You would prefer it without its own function?